client.go

```diff
@@ -597,6 +597,12 @@ func NewClient[TTx any](driver riverdriver.Driver[TTx], config *Config) (*Client
 		}
 	}
 
+	for _, hook := range config.Hooks {
+		if withBaseService, ok := hook.(baseservice.WithBaseService); ok {
+			baseservice.Init(archetype, withBaseService)
+		}
+	}
```

Comment:
I added some code to the client and test worker so that if hooks/middleware are base services, we'll initialize them. This isn't intended for external use, but makes it really convenient for sharing things like a logger so that those don't need to be configured into the hook/middleware separately.